File: lecture_007(Strings)/l_012_removeprimes.cpp
#include<iostream>
#include<vector>
using namespace std;

bool checkPrime(int n) {
	for(int div=2; div*div<=n; div++) {
    	if(n%div==0) {
    		return false;
		}
	}
	return true;
}
vector<int> removeprimes(vector<int> a1) {
    for(int i=a1.size()-1; i>=0; i--) {
		int val = a1[i];
    	if(checkPrime(val)==true) {
    		a1.erase(a1.begin()+i);
        }
    }
    return a1;
}
int main() {
    int n;
    cin>>n;
    vector<int> a1(n,0);
    for(int i=0; i<n; i++) {
        cin>>a1[i];
    }
    for(int i=0; i<n; i++) {
        cout<<a1[i]<<" ";
    }
    cout<<endl;
    vector<int> result = removeprimes(a1);
    for(int i=0; i<result.size(); i++) {
        cout<<result[i]<<" ";
    }
}

File: lecture_007(Strings)/l_012_removeprimes.java
import java.util.Scanner;
import java.util.ArrayList;
public class l_012_removeprimes {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scn.nextInt();
        ArrayList<Integer> a1 = new ArrayList<>();
        for(int i=0; i<n; i++) {
        	a1.add(scn.nextInt());
        }
        System.out.println(a1);
        System.out.println(removePrimes(a1));
    }
    public static boolean checkPrime(int n) {
    	for(int div=2; div*div<=n; div++) {
    		if(n%div==0) {
    			return false;
    		}
    	}
    	return true;
    }
    public static ArrayList<Integer> removePrimes(ArrayList<Integer> a1) {
    	for(int i=a1.size()-1; i>=0; i--) {
    		int val = a1.get(i);
    		if(checkPrime(val)==true) {
    			a1.remove(i);
    		}
    		
    	}
    	return a1;
    }
}